### PR TITLE
Feat : add delete caseStudies endpoint

### PR DESCRIPTION
**Pull request by rishusingh022**

### CODE REVIEW

File: src/middlewares/case-study.delete.validator.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+const Joi = require('joi');
```
Comment (rohini-senthil) [inline src/middlewares/case-study.delete.validator.js:1]: Can we have all case study validators in the same file as `case-study.validator.js`, same for user as well

Comment (rishusingh022) [reply src/middlewares/case-study.delete.validator.js]: done !!

File: src/controllers/case-study.controller.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
+const updateCaseStudyServices = require('../services/case-study.service');
```
Comment (rohini-senthil) [inline src/controllers/case-study.controller.js:1]: Can we keep this `caseStudyService`? It is not solely related only to update case study

File: src/middlewares/case-study.validator.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  name: Joi.string().min(3).max(30),
+  description: Joi.string(),
+  collaborators_ids: Joi.array().items(uuidType),
+  image: Joi.string(),
+  box_link: Joi.string(),
+  skills: Joi.array().items(Joi.string()),
+  engagement_id: Joi.array().items(uuidType),
+});
+
+const CaseStudyIdValidator = (req, res, next) => {
```
Comment (rohini-senthil) [inline src/middlewares/case-study.validator.js:23]: `CaseStudyIdValidator ` – This shouldn't be capitalized

**Review by rohini-senthil — CHANGES_REQUESTED** [review]
Requested changes

File: src/middlewares/case-study.validator.js (hunk 3 above)
Comment (rishusingh022) [reply src/middlewares/case-study.validator.js]: done !

File: src/controllers/case-study.controller.js (hunk 2 above)
Comment (rishusingh022) [reply src/controllers/case-study.controller.js]: done !!

File: src/controllers/case-study.controller.js [hunk 4]
```diff
@@ -14,6 +16,20 @@ const updateCaseStudyController = async (req, res) => {
   }
 };
 
+const deleteCaseStudyController = async (req, res) => {
```
Comment (rohini-senthil) [inline src/controllers/case-study.controller.js:19]: This doesn't need to be named "Controller" explicitly. Let's rename these methods to `updateCaseStudy` and `deleteCaseStudy`

File: src/middlewares/case-study.validator.js [hunk 5]
```diff
@@ -6,7 +6,7 @@ const uuidType = Joi.string().uuid({
   version: ['uuidv4', 'uuidv1'],
 });
 
-const updateCaseStudyIdSchema = Joi.object({
+const CaseStudyIdSchema = Joi.object({
```
Comment (rohini-senthil) [inline src/middlewares/case-study.validator.js:9]: Can we change this to camel case?

**Pull request merged**